assets/deepen_shallow_clone_until_ref_is_found

```diff
@@ -20,31 +20,30 @@ readonly ref="$1"
 # In either case we try to deepen the shallow clone until we find $ref, reach
 # the max depth of the repo, or give up after a given depth and resort to deep
 # clone.
-if [ "$concourse_git_resource__depth" -gt 0 ]; then
-  depth="$concourse_git_resource__depth"
-  git_dir="$(git rev-parse --git-dir)"
-  readonly git_dir
-
-  while ! git checkout -q "$ref" &>/dev/null; do
-    # once the depth of a shallow clone reaches the max depth of the origin
-    # repo, Git silenty turns it into a deep clone
-    if [ ! -e "$git_dir"/shallow ]; then
-      echo "Reached max depth of the origin repo while deepening the shallow clone, it's a deep clone now"
-      break
-    fi
-
-    echo "Could not find ref ${ref} in a shallow clone of depth ${depth}"
-
-    (( depth *= 2 ))
-
-    if [ "$depth" -gt "$max_depth" ]; then
-      echo "Reached depth threshold ${max_depth}, falling back to deep clone..."
-      git fetch --unshallow origin
-
-      break
-    fi
-
-    echo "Deepening the shallow clone to depth ${depth}..."
-    git fetch --depth "$depth" origin
-  done
-fi
+
+depth="$concourse_git_resource__depth"
+git_dir="$(git rev-parse --git-dir)"
+readonly git_dir
+
+while ! git checkout -q "$ref" &>/dev/null; do
+  # once the depth of a shallow clone reaches the max depth of the origin
+  # repo, Git silenty turns it into a deep clone
+  if [ ! -e "$git_dir"/shallow ]; then
+    echo "Reached max depth of the origin repo while deepening the shallow clone, it's a deep clone now"
+    break
+  fi
+
+  echo "Could not find ref ${ref} in a shallow clone of depth ${depth}"
+
+  (( depth *= 2 ))
+
+  if [ "$depth" -gt "$max_depth" ]; then
+    echo "Reached depth threshold ${max_depth}, falling back to deep clone..."
+    git fetch --unshallow origin
+
```

Comment:
You're missing `git checkout -q $ref` here - you'll unshallow the repo, but not check out the right commit, if you're in a submodule.

Reply:
Indeed. Fixed in 82ef5a393a1b1e5a00f5413175f10350800491e8 and ad553ecb6aa971840fb6be0b805a50616d2b2480.